sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java

```diff
@@ -109,41 +144,56 @@ public String version() {
         return KafkaCosmosConstants.CURRENT_VERSION;
     }
 
-    private List<Map<String, String>> getTaskConfigs(int maxTasks) {
-        Pair<MetadataTaskUnit, List<FeedRangeTaskUnit>> taskUnits = this.getAllTaskUnits();
+    private IMetadataReader getMetadataReader() {
+        switch (this.config.getMetadataConfig().getStorageType()) {
+            case KAFKA:
+                return this.kafkaOffsetStorageReader;
+            case COSMOS:
+                CosmosAsyncContainer metadataContainer =
```

Comment:
Should we do a read here to validate that the conatiner has valid config (like if you expect any certain PK + index policy)?

Reply:
yea make sense - also one question I have here is that should we create the metadata container if it does not exists? I did not add it because it will complicated the story for AAD auth etc.  But open for suggestions

Reply:
IMO better to provide a help function to create the container - I would never create it dynamically at runtime.

Reply:
will do in next PR